docs/reference/authentication.md

```diff
@@ -47,3 +47,32 @@ with the command `conjur authn whoami`.
 $ conjur authn whoami
 {"account":"myorg","username":"joe-tester"}
 {% endhighlight %}
+
+## `authn-local`
+
+If you provide your Conjur server with a directory `/run/authn-local`, Conjur will run a service called `authn-local` which listens on a Unix socket in this directory.
+
+You can send JSON requests to this socket. Each request must be formatted on a single line (no "pretty-printed" JSON). The JSON is a map which contains:
+
+* **account** The account for which the token should be issued (required).
+* **sub** The token subject (required).
+* **exp** The expiration time of the token (optional).
+* **cidr** A CIDR restriction on token validity (optional).
+
+The response will be a Conjur access token. For more details about access tokens, see [Cryptography / Authentication tokens](/reference/cryptography.html#authentication-tokens).
+
+`authn-local` is designed to be used by web services which provides custom authentication. Authenticators are responsible for implementing custom authentication logic, such as:
```

Comment:
Maybe this section and the following discussion of security considerations should move above the how-to discussion. Providing a clear idea of why you'd want to enable `authn-local`, and what the security implications are, feels like it should happen before encouraging the user to dive in.